src/langrila/memory/cosmos.py

```diff
@@ -0,0 +1,45 @@
+from azure.cosmos import exceptions, CosmosClient, PartitionKey
+import logging
+import os
+import uuid
+
+from ..base import BaseConversationMemory
+
+class CosmosConversationMemory(BaseConversationMemory):
+    def __init__(self, endpoint_name: str, key_name: str, dbname_name: str, containername_name: str, pkey_name: str):
+        self.endpoint = os.getenv(endpoint_name)
+        self.key = os.getenv(key_name)
+        self.dbname = os.getenv(dbname_name)
+        self.containername = os.getenv(containername_name)
+        self.pkey = '/' + os.getenv(pkey_name).strip('/')
+        # Create a Cosmos client
+        try:
+            client = CosmosClient(url=self.endpoint, credential=self.key)
+        except:
+            logging.error('Could not connect to Cosmos DB')
+            raise ConnectionError
+        # Get a database
+        try:
+            database = client.get_database_client(database=self.dbname)
+        except exceptions.CosmosResourceNotFoundError:
+            logging.error(f'Could not find database: {self.dbname}')
+        # Get a container
+        try:
+            self.container = database.get_container_client(self.containername)
+        except exceptions.CosmosResourceExistsError:
+            logging.error(f'Could not find container: {self.containername}')
+
+    def store(self, conversation_history: list[dict[str, str]]):
+        for item in conversation_history:
+            item['id'] = str(uuid.uuid4())
+            self.container.create_item(item)
+
+    def load(self) -> list[dict[str, str]]:
+        result = []
+        try:
+            items = self.container.read_items()
+            for item in items:
+                result.append(item)
```

Comment:
In later step needs only `role` and `content`. 

```suggestion
result.append({k: v for k, v in item.items() if k in {"role", "content"}})
```

Reply:
Cosmos DB returns not only 'role' and 'content' actually. e.g. 'id', '_rid', '_etag', '_ts', and so on.
Especially, '_ts', timestamp is important to sort the conversation history.

But I can also understand your intention to keep your langrila small and clean. Any ideas?

Reply:
Sorry, I don't quite understand your intention precisely yet. I'd appreciate it if you could give me more details.
How can we use those metadatas from CosmosDB? Or how will those metadatas be used at other place out of this module.

Basically I'm trying to close the responsibility of each module into each module itself. If we need to sort conversation memories with `_ts `and if that's all to do, it might be enough to use `_ts` for sorting memories within `load()` and to output the list of message dictionary including only `role` and `content`.
But I also think I couldn't understand due to my poor knowledge on CosmosDB. Thus I need more details.

Reply:
I understand your intention to close the responsibility. load() in CosmosConversationMemory should be implemented exactly same to one in JSONConversationMemory.
If a user needs to use other values from Cosmos DB, he/she can inherit CosmosConversationMemory class and add load_all() or something to get all the values from Cosmos. It's clean.